radar-commons-android/src/main/java/org/radarbase/android/data/TapeCache.kt

```diff
@@ -251,6 +257,35 @@ constructor(
         }
     }
 
+    /**
+     *  Comparing the size of queue with maximumSize.
+     *  Tolerance value is used to avoid the problems
+     *  with comparing floating point numbers for
+     *  equality due to rounding errors
+    */
+     fun checkStorageStatus(queueSize: Long) {
+        val threshold = maximumSize * 0.75
+        val size = queueSize.toDouble()
+        if (size >= threshold - 0.0001 && size <= threshold + 0.0001) {
+             setLevelPartial()
+        }
+        else{
+            setLevelAvailable()
+        }
+    }
+
+    private fun setLevelAvailable() {
+        storagelevelreceiver.setLevel(0.0f)
+    }
+
+    private fun setLevelPartial() {
+        storagelevelreceiver.setLevel(0.75f)
```

Comment:
By letting the calculation happen in the receiver, this simplifies a lot
```suggestion
        storagelevelreceiver.setLevel(queueSize.toFloat() / maximumSize.toFloat())
```